Add 'Planned Enhancements' section to DG

chiaryan opened the pull request:
As part of our post-PE Dry Run, we can document our product feature flaws that we cannot implement for v1.4 under a 'Planned Enhancements' section.

Let's add some features that will tackle some of the issues raised by our product testers for the PE.

Further validation of student details -- Closes #132, Closes #122, Closes #112, Closes #111
Enhanced Name Input -- Closes #143 
Further input validation for commands -- Closes #131, Closes #118

chiaryan committed:
Add 'Planned Enhancements' section to DG

As part of our post-PE Dry Run, we can document our product feature flaws that we cannot implement for v1.4 under a 'Planned Enhancements' section.

Let's add some features that will tackle some of the issues raised by our product testers for the PE.

chiaryan committed:
Merge branch 'master' into ug-future-enhancements

evedaktyl commented on docs/DeveloperGuide.md:
```diff
@@ -649,7 +649,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  * Phone numbers in Singapore should start with `8` or `9`.
+* `Email` addresses should follow the _local_@_domain_.com format.
+* Students should have unique contact details.
+  * This means that `add` or `edit` command will fail if it results in 
```
Incomplete

evedaktyl commented on docs/DeveloperGuide.md:
```diff
@@ -649,7 +649,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+Cons: Such additional constraints may restrict the user too much 
+
+**Alternative 1**
+* Have as toggleable 'Strict Mode' option that the user can enable of they want ClassMonitor to ensure that they cannot enter fields that do not fulfil the limitations.
```
typo at 'of'

evedaktyl commented on docs/DeveloperGuide.md:
```diff
@@ -649,7 +649,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+* Have as toggleable 'Strict Mode' option that the user can enable of they want ClassMonitor to ensure that they cannot enter fields that do not fulfil the limitations.
+* Otherwise, only limit fields by string length in 'Unstrict Mode'.
+
+### Enhanced Name Input
```
Capitalization for heading is inconsistent here, maybe we can standardize across headers.

evedaktyl commented on docs/DeveloperGuide.md:
```diff
@@ -649,7 +649,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+* Otherwise, only limit fields by string length in 'Unstrict Mode'.
+
+### Enhanced Name Input
+* Allow for ways to input names not are not normally allowed as command parameters, specifically the `/` character in 'S/O'
```
Seems like there is an extra 'not are'

evedaktyl commented on docs/DeveloperGuide.md:
```diff
@@ -649,7 +649,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+Command inputs can be further restricted to align with the field constraints
+* the `find` command can detect inputs that are out of range (e.g. `find star > 50000` will throw an error because ClassMonitor can never have more than 50000 stars) 
+
+### More granular error messages.
```
Remove full stop

evedaktyl reviewed (commented):
Very detailed and comprehensive section, left some suggestions to standardize the use of full stops and capitalization for headers.

chiaryan committed:
Minor Checkstyle Fix: Fix grammar, inconsistent Capitalization, periods

Fixes problems raised in
https://github.com/AY2324S2-CS2103T-F13-4/tp/pull/164#pullrequestreview-1997828300

chiaryan committed:
Merge master
